### PR TITLE
Migration & added functionalities

### PR DESCRIPTION
**Pull request by veroandi**
The branch "devMaster" contains the site's version before migrating to Symphony 3.2 . 
The devMigration contains the new version of the site, and the functionalities successfully implemented till now are:

- public pages (home, contact, mentions légales etc);
- inscription (the user is able to sign up as a simple user or professional);
- connexion (the user is able to sign-in);
- user's account: main page for particular & pro;
- user's account: user can view & change profile; 
- user's account: access to settings page (functionalities are not working yet);
- user's account: access to page with published reviews (not finished yet);

If user is Pro:
- access to view list of establishments;
- view & edit information of each establishment; 
- view the reviews of each establishment; (answers aren't working yet)

### CODE REVIEW

File: app/config/security.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            #form_login: ~
+
+    role_hierarchy:
+        ROLE_ADMIN: [ROLE_PART, ROLE_PRO]
+
+    access_control:
+        - { path: ^/, roles: IS_AUTHENTICATED_ANONYMOUSLY}
+        - { path: ^/login, roles: IS_AUTHENTICATED_ANONYMOUSLY}
+        #- { path: ^/dashboard, roles: IS_ATHENTICATED_REMEMBERED} il faudra changer pour ROLE_ADMIN
+        - { path: ^/, roles: ROLE_USER }
```
Comment (2binfree) [inline app/config/security.yml]: same path, different roles ???

File: src/CaradvisorBundle/Controller/MainController.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+use Symfony\Component\HttpFoundation\Request;
+
+class MainController extends Controller
+{
+    /**
+     * @Route("/", name="home")
+     */
+    public function indexAction()
+    {
+        $reviews = $this->getDoctrine()->getRepository('CaradvisorBundle:Review')->findAll();
```
Comment (2binfree) [inline src/CaradvisorBundle/Controller/MainController.php]: findall forbidden. You must use paginator

File: src/CaradvisorBundle/Controller/SecurityController.php [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            $user->setToken(null);
+            $user->setDateLimitToken(null);
+
+            $em = $this->getDoctrine()->getManager();
+            $em->persist($user);
+            $em->flush();
+
+            $this->addFlash("notice-green", "Votre compte à bien été enregistré, vous pouvez maintenant vous connecter.");
+        } else {
+            $this->addFlash("notice-red", "Votre compte n'a pas été enregistré, veuillez vous réinscrire.");
```
Comment (2binfree) [inline src/CaradvisorBundle/Controller/SecurityController.php]: Nous n'avons pas pu traiter votre demande.

File: src/CaradvisorBundle/Controller/UserController.php [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+    // Home page for user (professionals & non-professionals)
+    /**
+     * @Route("/user/{user}", name="user")
+     * @param User $user
+     * @return Response
+     */
+    public function indexAction(User $user)
+    {
+        /* Deny Access if not logged in
+        $this->denyAccessUnlessGranted(['ROLE_PART', 'ROLE_PRO'], null, 'Unable to access this page!'); */
```
Comment (2binfree) [inline src/CaradvisorBundle/Controller/UserController.php]: remove unnecessary  comments

File: src/CaradvisorBundle/Controller/UserController.php [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+     * @Route("/user/establishments/reviews/answer/{user}/{pro}", name="reviews_answer")
+     * @param Request $request
+     * @param User $user
+     * @param Pro $pro
+     * @param Answer $answer
+     * @return Response
+     */
+    public function answerReviewsAction(Request $request, User $user, Pro $pro, Answer $answer)
+    {
+       /* $answer = new Answer();
```
Comment (2binfree) [inline src/CaradvisorBundle/Controller/UserController.php]: remove unnecessary comments

File: src/CaradvisorBundle/Repository/ProRepository.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace CaradvisorBundle\Repository;
+
+class ProRepository extends \Doctrine\ORM\EntityRepository
+{
+
+    public function getReviewBuys()
+    {
+        $queryBuilder = $this->createQueryBuilder('p');
```
Comment (2binfree) [inline src/CaradvisorBundle/Repository/ProRepository.php]: unnecessary

File: src/CaradvisorBundle/Repository/ReviewBuyRepository.php [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+namespace CaradvisorBundle\Repository;
+
+use Doctrine\ORM\EntityRepository;
+
+class ReviewBuyRepository extends EntityRepository
+{
+    /**
+     * @return array
+     */
+    public function getReviewBuys($id)
```
Comment (2binfree) [inline src/CaradvisorBundle/Repository/ReviewBuyRepository.php]: use find

File: src/CaradvisorBundle/Resources/views/Default/home.html.twig [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+            <div class='row'>
+                <div class='col-md-offset-2 col-md-8'>
+                    <div class="carousel slide" data-ride="carousel" id="quote-carousel">
+                        <!-- Carousel Slides / Quotes -->
+                        <div class="carousel-inner">
+                            {% for review in reviews %}
+                            <!-- Quote 1 -->
+                            <div class="item {% if(review.id == 1) %}
+                                                     {{ "active" }}
+                                             {% endif %}">
```
Comment (2binfree) [inline src/CaradvisorBundle/Resources/views/Default/home.html.twig]: indentation

File: app/config/security.yml (hunk 1 above)
Comment (veroandi) [inline app/config/security.yml]: 👍 corrected!

File: src/CaradvisorBundle/Controller/MainController.php (hunk 2 above)
Comment (veroandi) [inline src/CaradvisorBundle/Controller/MainController.php]: 👍 corrected!

File: src/CaradvisorBundle/Controller/SecurityController.php (hunk 3 above)
Comment (veroandi) [inline src/CaradvisorBundle/Controller/SecurityController.php]: 👍 phrase was updated!

File: src/CaradvisorBundle/Controller/UserController.php (hunk 4 above)
Comment (veroandi) [inline src/CaradvisorBundle/Controller/UserController.php]: Done!

File: src/CaradvisorBundle/Controller/UserController.php (hunk 5 above)
Comment (veroandi) [inline src/CaradvisorBundle/Controller/UserController.php]: Done!

File: src/CaradvisorBundle/Repository/ProRepository.php (hunk 6 above)
Comment (veroandi) [inline src/CaradvisorBundle/Repository/ProRepository.php]: Done!

File: src/CaradvisorBundle/Repository/ReviewBuyRepository.php (hunk 7 above)
Comment (veroandi) [inline src/CaradvisorBundle/Repository/ReviewBuyRepository.php]: Not using file; code deleted

File: src/CaradvisorBundle/Resources/views/Default/home.html.twig (hunk 8 above)
Comment (veroandi) [inline src/CaradvisorBundle/Resources/views/Default/home.html.twig]: Done!

**Pull request merged**